js/print.js

```diff
@@ -0,0 +1,316 @@
+    }
+
+    if (secnum.textContent !== '') {
+      h1.insertBefore(document.createTextNode(' '), h1.firstChild);
+      h1.insertBefore(secnum, h1.firstChild);
+    }
+  });
+}
+
+/**
+ * The emu-imports element interferes with a ton of css specificity so we just
+ * take everything out and plop it directly into the DOM
+ * */
+function removeEmuImports() {
+  const emuImports = Array.from(specContainer.getElementsByTagName('emu-import'));
+
+  emuImports.forEach(importedMarkup => {
+    while (importedMarkup.hasChildNodes()) {
+      importedMarkup.parentNode.insertBefore(importedMarkup.childNodes[0], importedMarkup);
+    }
+    importedMarkup.parentNode.removeChild(importedMarkup);
+  });
+}
+
+/**
+ * Sets up table captions and figcaptions for tables, which provides for
+ * continuation table captions.
+ * */
+function rearrangeTables() {
+  const tables = Array.from(document.getElementsByTagName('emu-table'));
+
+  tables.forEach(emuTable => {
+    const figcaption = emuTable.getElementsByTagName('figcaption')[0];
+    const tableCaptionText = figcaption.innerHTML;
+    const table = emuTable.getElementsByTagName('table')[0];
+    const captionElement = document.createElement('caption');
+
+    captionElement.innerHTML = tableCaptionText;
+
+    table.insertBefore(captionElement, table.getElementsByTagName('thead')[0]);
+    table.appendChild(figcaption);
+  });
+}
+
+/**
+ * Gets rid of elements we don't need in the print version
+ * */
+function removeUnusedSections() {
+  const ecmaLogo = document.getElementById('ecma-logo');
+
+  if (ecmaLogo) specContainer.removeChild(ecmaLogo.parentNode);
+
+  document.getElementsByTagName('body')[0].removeChild(document.getElementById('shortcuts-help'));
+}
+
+function generateFrontCover() {
+  const frontCover = document.createElement('div');
+
+  shortname.innerHTML = shortname.innerHTML.replace(/standard/i, '');
+  // eslint-disable-next-line prettier/prettier
+  shortname.innerHTML = shortname.innerHTML.replace(/(draft|proposal)/i, '<span class="status">$1</span>');
+  version.innerHTML = version.innerHTML.replace(/(\d+)(st|nd|rd|th)/i, '$1<sup>$2</sup>');
+  title.innerHTML = title.innerHTML.replace(/(®|&reg;)/, '<sup>&reg;</sup>');
+
+  frontCover.classList.add('full-page-svg');
+  frontCover.setAttribute('id', 'front-cover');
+
+  frontCover.appendChild(shortname);
+  frontCover.appendChild(version);
+  frontCover.appendChild(title);
+
+  return frontCover;
+}
+
+function generateInsideCover() {
+  const insideCover = document.createElement('div');
+
+  insideCover.classList.add('full-page-svg');
+  insideCover.setAttribute('id', 'inside-cover');
+  insideCover.innerHTML =
+    '<p>Ecma International<br />Rue du Rhone 114 CH-1204 Geneva<br/>Tel: +41 22 849 6000<br/>Fax: +41 22 849 6001<br/>Web: https://www.ecma-international.org<br/>Ecma is the registered trademark of Ecma International.</p>';
+
+  return insideCover;
+}
+
+function generateEcmaCopyrightPage() {
```

Comment:
Why are we generating another copyright page? Don't we already have one in the document?

Reply:
The document has the original copyright but not the "alternative copyright”. Would extremely love to be proven wrong on this though, it’s quite possible I’m missing something with copyright minutae.